src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashCause.java

```diff
@@ -123,18 +127,38 @@ public String getBuildStartCommentId() {
 
   @Override
   public String getShortDescription() {
-    return "<a href=\""
-        + stashHost
-        + "/projects/"
-        + this.getDestinationRepositoryOwner()
-        + "/repos/"
-        + this.getDestinationRepositoryName()
-        + "/pull-requests/"
-        + this.getPullRequestId()
-        + "\" >PR #"
-        + this.getPullRequestId()
-        + " "
-        + this.getPullRequestTitle()
-        + " </a>";
+    return "<a href='" + getEscapedUrl() + "'>" + getEscapedDescription() + " </a>";
+  }
+
+  String getEscapedUrl() {
+    return getEncoder().encodeForHTMLAttribute(getPrUrl().toASCIIString());
+  }
+
+  String getEscapedDescription() {
+    return getEncoder()
+        .encodeForHTML("PR #" + this.getPullRequestId() + " " + this.getPullRequestTitle());
+  }
+
+  URI getPrUrl() {
+    try {
+      return new URI(stashHost)
+          .resolve(
+              new URI(
+                  null,
+                  null,
+                  "/projects/"
+                      + this.getDestinationRepositoryOwner()
+                      + "/repos/"
+                      + this.getDestinationRepositoryName()
+                      + "/pull-requests/"
+                      + this.getPullRequestId(),
+                  null));
+    } catch (URISyntaxException e) {
+      throw new IllegalStateException(e);
```

Comment:
sb-contrib detects an issue here: "Unconstrained method stashpullrequestbuilder.stashpullrequestbuilder.StashCause.getPrUrl() converts checked exception to unchecked [stashpullrequestbuilder.stashpullrequestbuilder.StashCause] At StashCause.java:[line 160] EXS_EXCEPTION_SOFTENING_NO_CONSTRAINTS"
Let's use checked exceptions, the compiler would force us to handle them.

Reply:
hiding the exception was intentional: my rationale was that it's unlikely or impossible to happen give the arguments